packages/schematics/angular/utility/ast-utils.ts

```diff
@@ -191,10 +191,7 @@ export function insertAfterLastOccurrence(nodes: ts.Node[],
                                           syntaxKind?: ts.SyntaxKind): Change {
   // sort() has a side effect, so make a copy so that we won't overwrite the parent's object.
   let lastItem = [...nodes].sort(nodesByPosition).pop();
-  if (!lastItem) {
-    throw new Error();
-  }
-  if (syntaxKind) {
+  if (syntaxKind && lastItem) {
```

Comment:
This still appears quite inefficient.  What about something like the following:
```
  let lastItem: ts.Node | undefined;
  for (const node of nodes) {
    if (!lastItem || lastItem.getStart() < node.getStart()) {
      lastItem = node;
    }
  }
  if (syntaxKind && lastItem) {
```
Efficiently handles the length 0 & 1 cases and avoids the extra array creation, sort mutations, and pop mutation.

Reply:
Maybe we should add this in a separate PR since the change emphasizes on/covers the insertion bug rather than improving the efficiency? What do you think?

Anyway, I agree with you that what you proposed is better performance-wise.

Reply:
Ok